rust/noosphere-core/src/data/address.rs

```diff
@@ -65,13 +74,88 @@ pub struct LinkRecord(Jwt);
 impl LinkRecord {
     /// Parse the wrapped [Jwt] as a [Ucan] and looks for the referenced pointer
     /// to some data in IPFS (via a [Cid] in the `fct` field).
-    pub async fn dereference(&self) -> Option<Cid> {
-        let token = &self.0;
-        let ucan = match Ucan::try_from(token.to_string()) {
-            Ok(ucan) => ucan,
-            _ => return None,
+    pub fn dereference(&self) -> Option<Cid> {
+        match Ucan::try_from(self) {
+            Ok(ucan) => LinkRecord::link(&ucan),
```

Comment:
A verb might be nice here e.g., `LinkRecord::find_link` or `extract_link` or something..